charts/gardener/controlplane/charts/application/templates/rbac-user.yaml

```diff
@@ -165,6 +171,14 @@ rules:
   - get
   - list
   - watch
+- apiGroup:
+  - apiextensions.k8s.io
+  resources:
+  - customresourcedefinitions
+  verbs:
+  - get
+  - list
+  - watch
```

Comment:
I think, we should be a bit more careful with granting watch permissions on the garden cluster.
However, we already have it for several other resources, that actually shouldn't need to be watched by end users, e.g. `apiservices`, `leases`, `events`, webhooks, etc.
How do we want to continue here?

Reply:
End users are usually using short-lived credentials and technically it will be hard for them to run a controller for more than several minutes/hours. So I am fine to keep the `watch` permissions as they are also valid option for read access.

Reply:
Frequent short-lived watches would be even worse...

Reply:
I think we should, as a matter of principle, grant `watch` to viewers, unless we are aware of actual issues caused by abuse of this permission and cannot deal with them in a different way. Do we have real examples?

Reply:
Well, on one of our larger enduser-facing landscapes I have seen the following memory usage of g-api:
```
$ kg top po -l app=gardener,role=apiserver
NAME                                                  CPU(cores)   MEMORY(bytes)
gardener-apiserver-dcbdbfb8-5n8lq                     1188m        13059Mi
gardener-apiserver-dcbdbfb8-lb8lc                     1366m        12191Mi
gardener-apiserver-dcbdbfb8-lkncr                     212m         11923Mi
gardener-apiserver-dcbdbfb8-m2n6s                     449m         11511Mi
gardener-apiserver-dcbdbfb8-s8v8s                     358m         13884Mi
gardener-apiserver-dcbdbfb8-xswtg                     272m         11542Mi
```

So far, we have not investigated this immense memory consumption.
But my current feeling is, that this is partly caused by endusers watching gardener resources. But this hypothesis still has to be proven.

I know, I'm basically hijacking this PR to bring up an issue that should be investigated separately and has been existing before this change.
But FMPOV this situation is already severe enough and adding more watch permissions (especially for extremely large resources like CRDs) will not relax the situation.
That's why I'm bringing this up now.

Reply:
Project members are already granted with `watch` access to most of the resources, see https://github.com/gardener/gardener/blob/5b63989418b1a02941fb305f44507b90a0d5f20e/charts/gardener/controlplane/charts/application/templates/rbac-user.yaml#L289-L387

They are also allowed to create their own `role` and `rolebindings`, so we cannot stop them from opening watch streams to the gardener API by excluding the `watch` verb from the clusterrole `gardener.cloud:viewer`. And this role by default is not bound to any subject, i.e. it is not actively used by default, instead gardener landscape administrators needs to explicitly define (cluster)rolebindings to it.

Based on this, I lean toward getting merging this change as it is today and tackle the high memory usage in separate issue/pr.

Reply:
> Based on this, I lean toward merging this change as it is today and tackle the high memory usage in separate issue/pr.

Fine for me 👍